build_rules.gradle

```diff
@@ -418,7 +418,7 @@ ext.applyJavaNature = {
   targetCompatibility = configuration.javaVersion
   tasks.withType(JavaCompile) {
     options.encoding = "UTF-8"
-    options.compilerArgs += ["-Xlint:all","-Xlint:-options","-Xlint:-cast","-Xlint:-deprecation","-Xlint:-processing","-Xlint:-rawtypes","-Xlint:-serial","-Xlint:-try","-Xlint:-unchecked","-Xlint:-varargs","-parameters"]
+    options.compilerArgs += ["-Xlint:all","-Xlint:-options","-Xlint:-cast","-Xlint:-deprecation","-Xlint:-dep-ann","-Xlint:-processing","-Xlint:-rawtypes","-Xlint:-serial","-Xlint:-try","-Xlint:-unchecked","-Xlint:-varargs","-parameters"]
```

Comment:
Should this be disabled?  From the [documentation](https://docs.oracle.com/javase/8/docs/technotes/tools/windows/javac.html):

> Warn about items that are documented with an @deprecated Javadoc comment, but do not have a @Deprecated annotation.

Seems like a useful warning. If we are missing a `@Deprecated` tag, I recommend adding it.

Reply:
Unfortunately there is a missing `@Deprecated` in the code generated via javacc. We are dependent on javacc 4.0, not sure the reasoning, nor do I know if upgrading would fix this issue and not cause others.

I'm honestly not really a fan of this solution either since it blankets the rest of the project. With the assumption that we keep javacc locked down, we could expand the `applyJavaNature` to have a param to enable this flag explicitly. Thoughts?

Reply:
Sorry for the slow response. I pulled your branch to investigate this a bit more. It seems that we are using a very old version of the javacc tool, and more recent versions don't have this issue.

https://github.com/cademarkegard/beam/pull/1 fixes it, but we should double-check with SQL contributors to ensure this won't break anything. /cc @apilloud @kennknowles

Reply:
thanks so much!